src/node_builtins.cc

```diff
@@ -234,23 +238,26 @@ void BuiltinLoader::AddExternalizedBuiltin(const char* id,
                 filename);
         ABORT();
       }
-      externalized_builtin_sources[id] = source;
+      size_t expected_u16_length =
+          simdutf::utf16_length_from_utf8(source.data(), source.length());
+      auto out = std::make_shared<std::vector<uint16_t>>(expected_u16_length);
+      size_t u16_length = simdutf::convert_utf8_to_utf16(
+          source.data(),
+          source.length(),
+          reinterpret_cast<char16_t*>(out->data()));
+      out->resize(u16_length);
+
+      auto resource_ptr = std::make_unique<StaticExternalTwoByteResource>(
+          out->data(), out->size(), out);
+      // OK to get the raw pointer, since externalized_builtin_sources owns
+      // the resource, resources are never removed from the map, and
+      // externalized_builtin_sources has static lifetime.
+      resource = resource_ptr.get();
+      externalized_builtin_sources[id] = std::move(resource_ptr);
```

Comment:
You can do

```suggestion
      auto result = externalized_builtin_sources.insert({
          id,
          std::make_unique<StaticExternalTwoByteResource>(out->data(), out->size(), out)
      });
      CHECK(result.second);
      it = result.first;
```

That we don't need the first branch - we can just do `resource = it->second.get()` after making sure `it` points to an actual entry.